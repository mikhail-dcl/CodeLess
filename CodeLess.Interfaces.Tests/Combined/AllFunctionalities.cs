using NUnit.Framework;

namespace CodeLess.Interfaces.Tests
{
    public class Compilation
    {
        [TestFixture]
        public class Combined
        {
            private const string TEST_DIR = nameof(Combined);

            [Test]
            public Task AllFunctionalities() =>
                InterfaceVerifier.RunAsync(TEST_DIR, nameof(AllFunctionalities));
        }
    }
}
