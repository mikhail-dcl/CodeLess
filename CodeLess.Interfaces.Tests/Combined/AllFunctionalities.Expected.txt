#nullable enable
using System;
using System.Threading.Tasks;

namespace CodeLess.Interfaces.Tests
{
    public partial interface IAllFunctionalities
    {
        [Obsolete]
        event Action SimpleEvent;
        event EventHandler<int> GenericEvent;
        event Action InitEvent;
        event EventHandler ExplicitEvent;
        [Obsolete("test")]
        event EventHandler AttributedEvent;
        /// <summary>Xml doc event</summary>
        event EventHandler XmlDocEvent;
        /// <summary>XML doc property</summary>
        string DocProp { get; set; }

        int ReadOnlyProp { get; }

        string WriteOnlyProp { set; }

        string InitOnlyProp { get; init; }

        string ExprBodiedProp { get; }

        int this[int idx] { get; set; }

        void Method();
        Task<int> AsyncMethod();
        TResult GenericMethod<T, TResult>(T input);
        void MethodWithParams(int a, string b, out bool ok, in double pi, ref long x, Action callback);
        (string, int) MethodOutInRefReturn(in string s, ref int count, out bool success);
    }
}